### PR TITLE
Basic `flagged` flag usage

### LINKED ISSUE #1464: Feature flags
## Feature Description

Have a method to have features merged into master but disable / hidden via a feature flag. 
This would allow developers to merge incomplete work while not showing it to users.

### PR DESCRIPTION
**Pull request by diegovar**
Integrates the `flagged` package and demonstrates its usage.

Next (optional) steps:

- [ ] Extract flags into json file and load from php
- [ ] Have different json files per environment
- [ ] Make JSON files inherit from each other (local inherits from test inherits from prod)
- [ ] Integrate into dashboard if necessary

### CODE REVIEW

**Comment by swissspidy** [comment]
Do we really need multiple inheriting config files per environment (yet)? Doesn‘t one suffice for now?

For testing & QA, I have been thinking that we could make a small settings screen in WordPress where features could be toggled.

**Comment by diegovar** [comment]
I agree that multiple inheriting config files per environment are totally not necessary, the next steps outlined are potential next steps.

The WP settings screen sounds like a great idea!

File: includes/Story_Post_Type.php
Comment (spacedmonkey) [inline includes/Story_Post_Type.php]: At some point, we could have a UI for this in WP, we can enable / disable flags and it persists.

Comment (swissspidy) [reply includes/Story_Post_Type.php]: Yep, that's what I mentioned in my comment too 👍

**Comment by spacedmonkey** [comment]
We could even have a little dialog in the editor it's, that saving the flags to in WordPress ( as options ) using the REST API. This maybe overkill. But it would allow users to opt in to beta features in the future.

**Comment by diegovar** [comment]
Do you think this PR is mergeable as a first step? It would unblock some downstream work.

File: assets/src/edit-story/index.js
Comment (swissspidy) [inline assets/src/edit-story/index.js:44]: Should this rather be called inside `<App>`?

Comment (spacedmonkey) [reply assets/src/edit-story/index.js:44]: I was going to ask this question

File: assets/src/edit-story/components/library/panes/media/mediaPane.js
Comment (swissspidy) [inline assets/src/edit-story/components/library/panes/media/mediaPane.js]: Before merge, this should probably be removed and instead added as an example in a new `/docs/feature-flags.md` document.

File: assets/src/edit-story/index.js
Comment (diegovar) [reply assets/src/edit-story/index.js:44]: the `flagged` sample code wraps the app completely: https://www.npmjs.com/package/flagged

But I guess it could go inside. I guess it's more in the case that you wanted to flag-guard anything in the App component itself. Thoughts?

File: assets/src/edit-story/components/library/panes/media/mediaPane.js
Comment (diegovar) [reply assets/src/edit-story/components/library/panes/media/mediaPane.js]: Done.

File: docs/feature-flags.md
Comment (swissspidy) [inline docs/feature-flags.md:1]: Mind linking to this in `/docs/README.md` under `Development`? That is basically the ToC for the docs there.

Thanks!

File: includes/Story_Post_Type.php
Comment (spacedmonkey) [inline includes/Story_Post_Type.php:401]: Should this load from say a json file?

Comment (swissspidy) [reply includes/Story_Post_Type.php:401]: That is mentioned as a possible next step in the PR description. So yes, maybe in the future we could do that.

File: docs/feature-flags.md
Comment (diegovar) [reply docs/feature-flags.md:1]: Done

**Review by swissspidy — APPROVED** [review]

**Review by spacedmonkey — APPROVED** [review]

File: assets/src/edit-story/index.js
Comment (dvoytenko) [reply assets/src/edit-story/index.js:44]: Main question: what happens when we don't have a `FlagsProvider` at all? Does it default to all flags are off?

Generally, I like this here - this helps with integration tests. Integration tests would likely to start from `app/index.js` and it'd be easy to wrap the whole integration test with the `<FlagsProvider flags={flags-I-am-currently-testing}>`. We could always pass flags around in args, but then we just increase the dependencies on them.

File: assets/src/edit-story/index.js
Comment (dvoytenko) [inline assets/src/edit-story/index.js:52]: Should we have a simpler model to turn the flags for development? E.g. pass them as an arg --flags and merge them into the final list?

**Review by dvoytenko — CHANGES_REQUESTED** [review]

File: assets/src/edit-story/index.js
Comment (diegovar) [reply assets/src/edit-story/index.js:44]: If you don't have a FlagsProvider, then the app fails (which is arguably better).

File: assets/src/edit-story/index.js
Comment (diegovar) [reply assets/src/edit-story/index.js:52]: Why is that easier than modifying the Story_Post_Type.php file? I think that would be necessary if/when this application runs outside of Wordpress.

File: assets/src/edit-story/index.js
Comment (dvoytenko) [reply assets/src/edit-story/index.js:44]: Ok. We can still deal with this on integration level, so I'm ok if this is here and not in `app/index.js` and slightly prefer that even.

File: assets/src/edit-story/index.js
Comment (dvoytenko) [reply assets/src/edit-story/index.js:52]: Maybe not much easier, but felt a bit simpler to me w/o requiring me to modify a source file and hoping not to accidentally commit it. The php file, though, is good for launches.

**Review by swissspidy — APPROVED** [review]

**Comment by swissspidy** [comment]
Should we also add this to the dashboard app while we're at it?

They have plenty of things that are half-way there but currently hidden (e.g. link to settings page).

/cc @carlos-kelly

File: assets/src/edit-story/index.js
Comment (cvolzke4) [reply assets/src/edit-story/index.js:44]: The story editor playground will need to be updated as well:
https://github.com/google/web-stories-wp/blob/master/.storybook/stories/playground/index.js#L55

Comment (swissspidy) [reply assets/src/edit-story/index.js:44]: Indeed! Could add some knobs to the storybook to easily toggle flags there

**Comment by diegovar** [comment]
> Should we also add this to the dashboard app while we're at it?
> 
> They have plenty of things that are half-way there but currently hidden (e.g. link to settings page).
> 
> /cc @carlos-kelly

I'd rather do this in a separate PR so this can be merged. I'll comment on the parent issue with more context.

**Review by dvoytenko — APPROVED** [review]

**Pull request merged**
